parse/TechsParser.cpp

```diff
@@ -105,8 +105,19 @@ namespace {
         std::vector<std::shared_ptr<Effect::EffectsGroup>> effectsgroups;
         if (kw.has_key("effectsgroups")) {
             boost::python::stl_input_iterator<effect_group_wrapper> effectsgroups_begin(kw["effectsgroups"]), effectsgroups_end;
-            for (auto it = effectsgroups_begin; it != effectsgroups_end; ++it)
-                effectsgroups.push_back(it->effects_group);
+            for (auto it = effectsgroups_begin; it != effectsgroups_end; ++it) {
+                const auto& effects_group = *it->effects_group;
+                effectsgroups.push_back(std::make_unique<Effect::EffectsGroup>(
```

Comment:
I was going to say that I suppose there should just be an `EffectsGroup::CloneUnique` as well, but you're pushing into a `vector<shared_ptr<...>>` so you could be using `make_shared` instead.

But there's also no need to wrap these in an pointer, since the `Tech` constructor just moves them out of the pointer anyway, so we could add a Tech constructor that takes `vector<EffectsGroup>` by value instead.

Reply:
and then probably `EffectsGroup(const EffectsGroup&)` should be implemented...

Reply:
I've changed to `make_shared` here.